Marlin/src/module/planner.cpp

```diff
@@ -1447,8 +1447,9 @@ void Planner::check_axes_activity() {
       }
     }
 
-    float t = autotemp_min + high * autotemp_factor;
-    LIMIT(t, autotemp_min, autotemp_max);
+    float atMin = (float)autotemp_min, atMax = (float)autotemp_max;
+    float t = atMin + high * autotemp_factor; // Do we need float precision here (slower) ?
+    LIMIT(t, atMin, atMax);
```

Comment:
The autotemp factor is stored as a float, so everything will be upcast to `float` in the calculation. To get rid of `float` here you will need to replace `autotemp_factor` —currently a small decimal number— with a scaled integer, and then down-scale the result of the multiplication.

Reply:
I can convert everything to celsius_t here and it'll be computed with 6 bits of precision for the small value (faster than float, since it's using HW integer multiplier). But the question is the range here. If `autotemp_factor` is very small we'll loose precision. How small is it ? Smaller than 0.15  (in that case, using float will be better) or much larger ?

Reply:
Users can set `autotemp_factor` to any legal `float` value, and its value should be preserved with a resolution of 0.001 to avoid annoying them.